### PR TITLE
adding gcp log compatibility and new status codes

### PR DESCRIPTION
**Pull request by markterrill**
Adds new status codes and a mqtt log format that actually works on GCP stack driver as otherwise the log_level wasn't actually showing warn/debug/info.

Relies on https://github.com/cesanta/mongoose-os/pull/598 for connecterror

Squashed:
added more event publishing
Added TCP connect timeout event
ignoring arg->fd as it can be 2!
assuming that arg->fd == 2 == error is a mistake. Have run debug showing normal logs can be sent as fd = 2. arg->level seems to be consistently used, errors going through as 0 as you'd expect. So will just use that!
added more event publishing

### CODE REVIEW

File: mos.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ config_schema:
   - ["mqtt1.server", ""]
   - ["debug.stdout_topic", "s", "", {title: "MQTT topic to publish STDOUT to"}]
   - ["debug.stderr_topic", "s", "", {title: "MQTT topic to publish STDERR to"}]
+  - ["mqtt.debug_use_log_level", "b", false, {title: "Use the cs_log_level enum instead of stdout/stderr flag"}]
```
Comment (rojer) [inline mos.yml]: please move this up to where new defintions are, after ws_path

File: src/mgos_mqtt.c [hunk 2]
```diff
@@ -176,6 +185,9 @@ bool mgos_mqtt_init(void) {
                                     mgos_sys_config_get_mqtt1());
   }
 
+  LOG(LL_DEBUG, ("mqtt log_level %d | debug_use_log_level %d", 99,
```
Comment (rojer) [inline src/mgos_mqtt.c]: please remove, it's not useful.

**Review by rojer — CHANGES_REQUESTED** [review]
LG overall, minor notes.
please format with clang-format to keep the style consistent.

**Comment by markterrill** [comment]
Hi,
Done!
Removed debug (which I found a reassuring confirmation), applied clang_format to that mgos_mqtt file, and reordered the new mqtt config flag.
Mark

File: mos.yml (hunk 1 above)
Comment (markterrill) [reply mos.yml]: Done

File: src/mgos_mqtt.c (hunk 2 above)
Comment (markterrill) [reply src/mgos_mqtt.c]: Done

File: mos.yml [hunk 3]
```diff
@@ -35,13 +35,15 @@ config_schema:
   - ["mqtt.max_queue_length", "i", 5, {title: "Maximum queue length for buffering QoS 1+ messages. 0 to disable queue."}]
   - ["mqtt.ws_enable", "b", false, {title: "Enable WebSocket encapsulation"}]
   - ["mqtt.ws_path", "s", "/mqtt", {title: "Path to use for WebSocket handshake"}]
-  # Alternative MQTT configuration. If enabled, client will alternate between mqtt and mqtt1
+  - ["mqtt.debug_use_log_level", "b", false, {title: "Use the cs_log_level enum instead of stdout/stderr flag"}]
+# Alternative MQTT configuration. If enabled, client will alternate between mqtt and mqtt1
```
Comment (rojer) [inline mos.yml]: please keep indent on this one

**Comment by rojer** [comment]
one tiny comment and we're good to go

**Comment by rojer** [comment]
thank you!

**Pull request merged**